assertj-core/src/test/java/org/assertj/core/api/throwable/ThrowableAssert_hasCause_Test.java

```diff
@@ -37,11 +37,22 @@ protected void verify_internal_effects() {
 
   @Test
   void should_fail_if_actual_and_expected_cause_have_different_types() {
-    assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> assertThat(new IllegalArgumentException(new IllegalStateException())).hasCause(new NullPointerException()))
-                                                   .withMessage(format("%n" +
-                                                                       "Expecting a cause with type:%n" +
-                                                                       "  \"java.lang.NullPointerException\"%n" +
-                                                                       "but type was:%n" +
-                                                                       "  \"java.lang.IllegalStateException\"."));
+    // GIVEN
+    Throwable actual = new IllegalArgumentException(new IllegalStateException());
+    Throwable expectedCause = new NullPointerException();
+
+    // WHEN
+    ThrowableAssert.ThrowingCallable assertAction = () -> assertThat(actual).hasCause(expectedCause);
+
+    // THEN
+    assertThatExceptionOfType(AssertionError.class)
+                                                   .isThrownBy(assertAction)
+                                                   .withMessageStartingWith(format("%n" +
+                                                                                   "Expecting a cause with type:%n" +
+                                                                                   "  \"java.lang.NullPointerException\"%n" +
+                                                                                   "but type was:%n" +
+                                                                                   "  \"java.lang.IllegalStateException\".%n" +
+                                                                                   "Throwable that failed the check:%n" +
+                                                                                   "\"java.lang.IllegalStateException%n"));
```

Comment:
the throwable that failed the check is an `IllegalArgumentException`, not the `IllegalArgumentException` cause.